main.sh

```diff
@@ -1,12 +1,40 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
 # set environment variables from .env
 set -a
 source .env
 set +a
 
-if [ "$MODE" == "deploy" ]; then
-    gunicorn PLATER.services.app:APP -b 0.0.0.0:2304 -w 4 -k uvicorn.workers.UvicornWorker
-else
-    uvicorn PLATER.services.app:APP --host 0.0.0.0 --port 9747 --reload
-fi
+function webServer {
+    echo "Starting web server ... "
+    if [ "$MODE" == "deploy" ]; then
+        gunicorn PLATER.services.app:APP -b ${WEB_HOST}:${WEB_PORT} -w 4 -k uvicorn.workers.UvicornWorker
+    else
+        uvicorn PLATER.services.app:APP --host ${WEB_HOST} --port ${WEB_PORT} --reload
+    fi
+}
+
+function heartbeat {
+    if [[ ! -z "${AUTOMAT_HOST}" ]]; then
+        echo "Contacting Automat"
+        testEndpoint=${PLATER_SERVICE_ADDRESS}:${WEB_PORT}/predicates
+        response=$(curl --write-out %{http_code} --silent --output /dev/null ${testEndpoint})
+        until [ $response = "200" ]; do
+            response=$(curl --write-out %{http_code} --silent --output /dev/null ${testEndpoint})
+            echo "Plater not ready sleeping... "
+            sleep 1
+        done
+        echo "Plater ready starting heartbeat ..."
+        python PLATER/services/heartbeat.py -a ${AUTOMAT_HOST}
+    fi
+    $*
+}
+
+# Run the web server in the background
+webServer &
+webServerPID=$!
+# Run heartbeat sender in background, only runs if AUTOMAT_SERVER env is provided
+heartbeat &
```

Comment:
I think that all of the heartbeat stuff belongs in the automat repo. It looks like we can reduce it to a single bash script, and include documentation in automat about how to use it to deploy your plater, instead of the basic plater deployment that lives in this repo.

Reply:
I think the benefit of having this script here is that we can do automatic registration to automat. If we don't have that here I think we would have to do manual service discovery from automat side. I was hoping to have as little intervention as possible when registering a plater into automat. Maybe I missing something but I don't see a way of automating that with out Plater initiating communication.

Reply:
I did not explain well. The heartbeat will still be deployed with the plater, but the code used to do so will live in the automat repo. This way there is just a one-way dependency between these repositories. _If_ you want to use automat, you have deploy your plater with the special script that automat provides. In this sense, plater is to automat as docker is to docker-compose. Docker-compose has to be aware of docker, but docker does not have to be aware of docker-compose.

Reply:
Thanks for the clarification, makes sense with avoiding repo dependencies. I think the thing to do would be extending the docker image of plater to include the heart beat stuff in automat and having a new plater image there. All the heartbeat stuff would also live there.